validator/validators/performance/performance_validator.go

```diff
@@ -0,0 +1,59 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package performance
+
+import (
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/awsservice"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/aws/amazon-cloudwatch-agent-test/validator/models"
+)
+
+type PerformanceValidator struct {
+	vConfig models.ValidateConfig
+}
+
+var _ models.ValidatorFactory = (*PerformanceValidator)(nil)
+
+func NewPerformanceValidator(vConfig models.ValidateConfig) models.ValidatorFactory {
+	return &PerformanceValidator{
+		vConfig: vConfig,
+	}
+}
+
+func (s *PerformanceValidator) GenerateLoad() (err error) {
+	var (
+		agentCollectionPeriod = s.vConfig.GetAgentCollectionPeriod()
+		agentConfigFilePath   = s.vConfig.GetCloudWatchAgentConfigPath()
+		dataType              = s.vConfig.GetDataType()
+		dataRate              = s.vConfig.GetDataRate()
+		receiver              = s.vConfig.GetPluginsConfig()
+	)
+	switch dataType {
+	case "logs":
+		err = common.StartLogWrite(agentConfigFilePath, agentCollectionPeriod, dataRate)
```

Comment:
With a name like "StartLogWrite", I assume it means it JUST starts the generator, but does not WAIT for it.
In other words I expect it to start generating in goroutines and immediately return.

Is this the case? Or does this call block?

Reply:
The name are confusing. However, `StartLogWrite` will wait for the generator to completely [generate the logs with X minutes](https://github.com/aws/amazon-cloudwatch-agent-test/blob/main/internal/common/logs.go#L18-L43)